go/vt/dbconnpool/connection.go

```diff
@@ -18,19 +18,38 @@ package dbconnpool
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"vitess.io/vitess/go/mysql"
 	"vitess.io/vitess/go/mysql/sqlerror"
+	"vitess.io/vitess/go/pools/smartconnpool"
 	"vitess.io/vitess/go/sqltypes"
 	"vitess.io/vitess/go/vt/dbconfigs"
 )
 
+type PooledDBConnection = smartconnpool.Pooled[*DBConnection]
+
 // DBConnection re-exposes mysql.Conn with some wrapping to implement
 // most of PoolConnection interface, except Recycle. That way it can be used
 // by itself. (Recycle needs to know about the Pool).
 type DBConnection struct {
 	*mysql.Conn
+	info dbconfigs.Connector
+}
+
+var errSettingNotSupported = errors.New("TODO")
```

Comment:
nit: probably add a better error message, most likely a bug if this is getting called.

Reply:
Fixed!